AppSearchSample/app/src/main/java/com/android/example/appsearchsample/NoteListItemAdapter.kt

```diff
@@ -43,31 +52,44 @@ class NoteListItemAdapter(private val onDelete: (Note?) -> Unit) :
   }
 
   override fun onBindViewHolder(holder: NoteViewHolder, position: Int) {
-    holder.bind(getItem(position), onDelete)
+    holder.bind(getItem(position), onDelete, position)
   }
 
   /** ViewHolder for [NoteListItemAdapter]. */
-  class NoteViewHolder(
+  inner class NoteViewHolder(
     binding: ItemNoteBinding,
   ) : RecyclerView.ViewHolder(binding.root) {
     private val noteTextView = binding.noteText
     private val noteDeleteButtonView = binding.noteDeleteButton
 
-    fun bind(note: Note?, onDelete: (Note?) -> Unit) {
-      noteTextView.text = note?.text
-      noteDeleteButtonView.setOnClickListener { onDelete(note) }
+    fun bind(sr: SearchResult, onDelete: (SearchResult?) -> Unit, position: Int) {
+      val note = sr.genericDocument.toDocumentClass(Note::class.java)
+      val sb = SpannableStringBuilder(note.text)
+
+      sr.matchInfos.forEach {
+        sb.setSpan(StyleSpan(BOLD), it.exactMatchRange.start, it.exactMatchRange.end, Spannable.SPAN_INCLUSIVE_INCLUSIVE)
```

Comment:
Given we will not be inserting any text to the span, and just highlighting an existing segment, should the ```SPAN_INCLUSIVE_INCLUSIVE``` flag be changed to ```SPAN_EXCLUSIVE_EXCLUSIVE``` ?